src/core/unit.c

```diff
@@ -4492,6 +4495,9 @@ void unit_warn_if_dir_nonempty(Unit *u, const char* where) {
         assert(u);
         assert(where);
 
+        if (!unit_log_level_test(u, LOG_NOTICE))
+                return;
```

Comment:
here, too, please change the log_struct() call below to use log_unit_struct(), too